core/query/coverage.py

```diff
@@ -33,8 +33,19 @@ def add_coverage_for_identifiers_chain(
             _db, (p[0] for p in parent_ids)
         )
         records = []
-        for equiv in equivs:
 
+        # Make sure we haven't already added this record to the session, but not committed it yet
```

Comment:
Why do we need this, if the method is not recursively calling itself?

Reply:
We need this, because this listener gets fired twice, once for Equivalency (input = 1, output = 2) and once for Equivalency (input = 2, output = 1).

This used to happen in the old listeners as well, but in the case of the old listeners, the add_for function wouldn't add a second record, because it had already been flushed, so it just returns the existing row.

In this case, its happening as part of the flush operation, so the new objects are not flushed on creation, but at the end all the listeners firing, so the new object can't be queried from the DB yet, so we happily create multiple copies, and fail later in the flush when we try to add them to the DB.

Reply:
Oooh thats an interesting problem to have